### ISSUE TITLE
Datastores inside DS-Cluster will not found if Cluster is inside a Folder

### ISSUE THREAD

**Issue by arogarth** [pryorda/vmware_exporter#64]
We have the Following Structure:

```
+ VCENTER
  + Datacenter
    + Folder
      + DS-Cluster
        + Datastore
```

The Datastores are not inside ´inventory` variable (Line 304 - vmware_exporter.py)

**Comment by arogarth** [comment]
Problem in Code... Inventory is build flat over folders and clusters... But not recursive... Try to fix it....

**Comment by Jc2k** [comment]
I thought thats what was happening to cause the error in your last issue.

The proper fix is probably to use PropertyCollector (see `fetch_batch_properties`) rather than traversing the folders our self. Not only is it more robust for things like your setup, but its faster too. Thats was already on my medium term todo list. It's easy to generate the list of hosts, but need to see how to get the cluster name etc when using this method without incurring extra magic API calls from pyvmomi.

**Comment by arogarth** [comment]
I created a Pull request #66 where datastore and hosts will be collect recursive over folder. This works on our setup.

**Comment by pryorda** [comment]
Nice. I think @jc2k made some headway on this. Check out the gitter room. The button is in the readme.

**Comment by pryorda** [comment]
@arogarth,@jc2k This was mostly fixed by #67, but should the other #63 be merged before we close this?

**Comment by pryorda** [comment]
#108 Should fix this
